test/metabase/query_processor_test/nested_queries_test.clj

```diff
@@ -75,19 +78,21 @@
 (defn breakout-results [& {:keys [has-source-metadata? native-source?]
                             :or   {has-source-metadata? true
                                    native-source?       false}}]
-  {:rows [[1 22]
-          [2 59]
-          [3 13]
-          [4  6]]
-   :cols [(cond-> (qp.test/breakout-col (qp.test/col :venues :price))
-            native-source?
-            (-> (assoc :field_ref [:field "PRICE" {:base-type :type/Integer}]
-                       :effective_type :type/Integer)
-                (dissoc :description :parent_id :nfc_path :visibility_type))
-
-            (not has-source-metadata?)
-            (dissoc :id :semantic_type :settings :fingerprint :table_id :coercion_strategy))
-          (qp.test/aggregate-col :count)]})
+  ;; Oracle driver returns base and effective type as Decimal, other drivers as Integer.
+  (let [field-type (case driver/*driver* :oracle :type/Decimal :type/Integer)]
```

Comment:
What's the reason this surfaces in this PR?

Reply:
I've wanted to extend `card-id-native-source-queries-test` to query not only h2, but also other dbms.

To the best of my knowledge, when `mt/test-drivers` is not used, h2 database is queried by default, even if it is not in the set of drivers currenly tested.

So why extending the test? Because there are still some drivers using honeysql 1. And comment removal from native card query has different code path, depending on used honeysql version.

Extenting the test this way requires:
1. Getting the right database name for the native query in the test code. (Current bigquery and snowflake failures)
2. Adjusting the `breakout-results` to handle small differences between drivers.

Those being different field types for oracle, and field name formatting.

---

Actually while writing this, I've realized:
1. how to avoid this `case *driver*` snippet completely,
2. how to make the test runnable even for bigquery and snowflake (hopefully, as I have no way to verify that locally). 

I will push the update shortly.